src/Aquifer.Jobs/Program.cs

```diff
@@ -1,34 +1,47 @@
 using Aquifer.Common.Clients;
 using Aquifer.Common.Clients.Http.IpAddressLookup;
+using Aquifer.Common.Jobs;
 using Aquifer.Common.Services;
 using Aquifer.Data;
 using Aquifer.Jobs.Clients;
 using Aquifer.Jobs.Configuration;
+using Aquifer.Jobs.Services;
 using Microsoft.Azure.Functions.Worker;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
 
-var host = new HostBuilder().ConfigureFunctionsWorkerDefaults()
+var host = new HostBuilder()
+    .ConfigureFunctionsWorkerDefaults()
     .ConfigureServices((context, services) =>
     {
         services.AddOptions<ConfigurationOptions>().Bind(context.Configuration);
 
-        var connectionString = context.Configuration.GetConnectionString("BiblioNexusDb");
+        var configuration = context.Configuration.Get<ConfigurationOptions>()
+            ?? throw new InvalidOperationException($"Unable to bind {nameof(ConfigurationOptions)}.");
+
+        services.AddDbContext<AquiferDbContext>(options => options
+            .UseSqlServer(configuration.ConnectionStrings.BiblioNexusDb, providerOptions => providerOptions.EnableRetryOnFailure(3))
+            .EnableSensitiveDataLogging(sensitiveDataLoggingEnabled: configuration.IsDevelopment)
+            .UseQueryTrackingBehavior(QueryTrackingBehavior.NoTracking));
+
+        services.AddApplicationInsightsTelemetryWorkerService();
+        services.ConfigureFunctionsApplicationInsights();
+        services.AddAzureClient(context.Configuration.Get<ConfigurationOptions>()!.IsDevelopment);
+
+        services.AddQueueServices(configuration.ConnectionStrings.AzureStorageAccount);
 
         services.AddSingleton<IAquiferAppInsightsClient, AquiferAppInsightsClient>();
-        services.AddSingleton<IAzureClientService, AzureClientService>();
         services.AddSingleton<IAquiferApiManagementClient, AquiferApiManagementClient>();
         services.AddHttpClient<IIpAddressLookupHttpClient, IpAddressLookupHttpClient>();
         services.AddSingleton<IAzureKeyVaultClient, AzureKeyVaultClient>();
-        services.AddSingleton<ISendGridClient, SendGridClient>();
-        services.AddApplicationInsightsTelemetryWorkerService();
-        services.ConfigureFunctionsApplicationInsights();
-        services.AddAzureClient(context.Configuration.Get<ConfigurationOptions>()!.IsDevelopment);
-        services.AddDbContext<AquiferDbContext>(options =>
-            options.UseSqlServer(connectionString, providerOptions => providerOptions.EnableRetryOnFailure(3)));
+        services.AddSingleton<IEmailService, EmailService>();
+        services.AddKeyedSingleton<IEmailService, SendGridEmailService>(nameof(SendGridEmailService));
```

Comment:
The `EmailSubscriber` is the only class that needs to use the `SendGridEmailService` implementation of `IEmailService`.  That said, if we ever need to bypass the queue or want to move away from that strategy the `EmailService` and `SendGridEmailService` both use the same interface.